tutor/src/components/exercises/details.jsx

```diff
@@ -28,36 +26,33 @@ class ExerciseDetails extends React.Component {
     selectedSection:       PropTypes.instanceOf(ChapterSection),
     displayFeedback:       PropTypes.bool,
     onSectionChange:       PropTypes.func,
-    topScrollOffset:       PropTypes.number,
     windowImpl:            PropTypes.object,
   };
 
-  static defaultProps = {
-    topScrollOffset: 40,
-  };
-
-  scroller = new ScrollTo({
-    windowImpl: this.props.windowImpl,
-    onAfterScroll: this.onAfterScroll,
-  });
-
   @observable currentIndex;
   @observable currentSection;
 
   @computed get exercises() {
     return this.props.exercises.array;
   }
 
-  componentDidMount() {
-    // this.scroller.scrollToSelector('.exercise-controls-bar');
-  }
-
   UNSAFE_componentWillMount() {
     const { selectedExercise } = this.props;
     if (selectedExercise) {
       this.currentIndex = findIndex(this.exercises, selectedExercise);
       if (-1 == this.currentIndex) this.currentIndex = 0;
-      this.currentSection = selectedExercise.page.chapter_section.asString;
+      this.currentSection = selectedExercise.page.chapter_section;
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!this.props.selectedSection.eq(prevProps.selectedSection)) {
+      const index = this.exercises.findIndex(
+        ex => ex.page.chapter_section.eq(this.props.selectedSection)
+      );
+      if (index != -1) {
+        this.currentIndex = index;
```

Comment:
Shouldn't `currentIndex` just be a computed property?

Reply:
nah, what this component does is page through an array of exercises.  it's pretty old though, what we'd probably do now is create a UX model that'd control the state